### PR TITLE
[TASK] Move logic for writing guides.xml into a separate class

### PR DESCRIPTION
**Pull request by brotkrueml**
Additionally, require "ext-dom" in the root composer.json to avoid complaints of missing extension in PhpStorm.

### CODE REVIEW

File: packages/typo3-guides-cli/src/Repository/Exception/FileException.php [hunk 1]
```diff
@@ -26,6 +26,16 @@ public static function notReadable(string $filePath): self
         );
     }
 
+    public static function notWritable(string $filePath): self
```
Comment (linawolf) [inline packages/typo3-guides-cli/src/Repository/Exception/FileException.php:29]: Do we have to introduce our own exceptions for this? It seems like a common issue to me?

File: packages/typo3-guides-cli/src/Repository/GuidesSettingsRepository.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class GuidesSettingsRepository
+{
+    /**
+     * @return int Number of bytes written
+     */
+    public function create(string $filePath, \DomDocument $document): int
+    {
+        // An error is suppressed to avoid PHP warnings when the file does
+        // not exist. As we check the return value later this should be okay.
+        $result = @file_put_contents($filePath, $document->saveXml());
```
Comment (linawolf) [inline packages/typo3-guides-cli/src/Repository/GuidesSettingsRepository.php:18]: Can't we use some library for writing files that offers all this?

File: packages/typo3-guides-cli/src/Repository/Exception/FileException.php (hunk 1 above)
Comment (brotkrueml) [reply packages/typo3-guides-cli/src/Repository/Exception/FileException.php:29]: I like to move the instantiation of an exception into the exception itself (with named constructors). This way it is cleaner and the throwing class does not have to deal with wordings.

File: packages/typo3-guides-cli/src/Repository/GuidesSettingsRepository.php (hunk 2 above)
Comment (brotkrueml) [reply packages/typo3-guides-cli/src/Repository/GuidesSettingsRepository.php:18]: Hmm, I will postpone this PR and handle the XML stuff first (which is far more complex and with more code in the command). Then we will see, if this PR is necessary in that way.

**Comment by brotkrueml** [comment]
Closed in favour of #251.

**Pull request closed without merging**